Website Redesign

Holzhaus opened the pull request:
The website started look a bit dated, so I think it's time for a facelift. I redesigned the page using the minimal [Barebones CSS framework](https://acahir.github.io/Barebones/) while trying to keep the "Mixxx look" intact (dark/orange color scheme).

**Before:**
![2020-05-31-02-51-44_1366x768](https://user-images.githubusercontent.com/1834516/83342099-609ae680-a2eb-11ea-9299-92ad02897ce6.png)
![2020-05-31-02-52-09_1366x768](https://user-images.githubusercontent.com/1834516/83342089-4fea7080-a2eb-11ea-9811-0538bde81f42.png)

**After:**
![2020-05-31-02-51-52_1366x768](https://user-images.githubusercontent.com/1834516/83342090-524cca80-a2eb-11ea-9e1d-1d6fe537248d.png)
![2020-05-31-02-52-13_1366x768](https://user-images.githubusercontent.com/1834516/83342087-4d881680-a2eb-11ea-9a5d-ffdf208a2d14.png)

Let me know what you think.

Holzhaus committed:
Redesign website based on Barebones CSS

URL: https://acahir.github.io/Barebones/

Holzhaus committed:
static/css/custom: Fix CSS formatting and use Mixxx colorscheme

Holzhaus committed:
static/css/custom: Add more formatting fixes and cleanups

Holzhaus committed:
Add responsive navbar menu with dropdown

Holzhaus committed:
Fix paddings and use correct container classes

Holzhaus committed:
Add cool CSS transitions for hamburger menu

Holzhaus committed:
pages/index: Improve news section on home page

Holzhaus committed:
pages/press: Move "About" to separate page and improve press mirror

Holzhaus committed:
pages/contact: Clean up contact page

Holzhaus committed:
static/css/custom: Increase feature icon size

Holzhaus committed:
pages/download: Improve download page

Holzhaus committed:
pages/news: Improve news page

Holzhaus committed:
static/css/custom: Remove some example styles

Holzhaus committed:
Add splash background to homepage

Holzhaus committed:
Add self-hosted raleway fonts

Holzhaus committed:
pages/get-involved: Improve "Get Involved" page

Holzhaus committed:
pages/support: Improve "Support & Community" page

Swiftb0y commented:
Stylewise, I have a couple of suggestions:
 * I personally prefer the the menu bar at the top of the screen before the "splash image". I'd like to see if others feel the same way.
 * In case we do eventually get an image slider, it should take up the entire splash image area imo. though we'd have to find a location to put the "DJ your way".
 * The free download button seems redundant when the link to the download page is right next to it in the header. I'd propose to get rid of the free download button and instead highlight the download link in the header instead. Maybe by using orange as the backdrop for that item. similar to how its done on the rekordbox page:
![image](https://user-images.githubusercontent.com/12380386/83353246-7f33c880-a351-11ea-82a6-59b9c240642c.png)

Holzhaus commented:
Thanks for the feedback.

> * I personally prefer the the menu bar at the top of the screen before the "splash image". I'd like to see if others feel the same way.

This only applies to the index page, right? I think I like it like this, but I was thinking about adding a bigger Mixxx logo to the splash, too. The menu bar is sticky, so as soon as you scroll down it will stay at the top of the screen.

> * In case we do eventually get an image slider, it should take up the entire splash image area imo. though we'd have to find a location to put the "DJ your way".

Yeah, but we must be careful not make the splash too big so that it doesn't fit vertically on wide screens. I was also thinking about adding a bigger Mixxx logo to the splash.

> * The free download button seems redundant when the link to the download page is right next to it in the header. I'd propose to get rid of the free download button and instead highlight the download link in the header instead. Maybe by using orange as the backdrop for that item. similar to how its done on the rekordbox page:
>   ![image](https://user-images.githubusercontent.com/12380386/83353246-7f33c880-a351-11ea-82a6-59b9c240642c.png)

True. I'll check if I can integrate it into the navbar.

Swiftb0y commented:
> The menu bar is sticky, so as soon as you scroll down it will stay at the top of the screen.

Yes. I saw that (by previewing staging.mixxx.org) and I like it a lot.

> I think I like it like this

What about starting a poll on zulip and using whatever the majority prefers?

> Yeah, but we must be careful not make the splash too big so that it doesn't fit vertically on wide screens. I was also thinking about adding a bigger Mixxx logo to the splash.

 +1 for a bigger logo definitely, I'm just a bit unsure about the splash. Having a splash definitively helps to create a better first impression, but the current design with the little deere preview is not great. What if we made the splash even thinner, and add the slider (or rather the deere image placeholder) just below it? this way, we have a splash and quickly give the user a peek into the mixxx UI.

> True. I'll check if I can integrate it into the navbar.

Great, in case we decide to move the navbar further away, we can keep the free download button as well.

Holzhaus committed:
pages/download: Clean up download page

Holzhaus committed:
pages/contact: Add facelift for the contributor list

Holzhaus committed:
pages/error: Fix error page formatting

Holzhaus committed:
pages: Add headline link anchors to multiple pages

Holzhaus committed:
pages/press: Clean up press page

Holzhaus committed:
templates/donate_paypal: Redesign PayPal button (and don't leak user data!)

Holzhaus committed:
pages/index: Add Mixxx Logo to splash

Holzhaus committed:
pages: Remove obsolete css classes

Holzhaus committed:
pages: Rename feature_icon css class to feature-icon

Holzhaus committed:
navbar: Integrate download button into menu bar

Holzhaus commented:
> > Yeah, but we must be careful not make the splash too big so that it doesn't fit vertically on wide screens. I was also thinking about adding a bigger Mixxx logo to the splash.
> 
> +1 for a bigger logo definitely, I'm just a bit unsure about the splash. Having a splash definitively helps to create a better first impression, but the current design with the little deere preview is not great.

Do you dislike a screenshot in the splash in general? Or do you think the small deere splash screen doesn't look good? Because I'd agree that it looks too crowded, but I think that would be resolved with a cleaner 2-deck LateNight Facelift III screenshot.

> What if we made the splash even thinner, and add the slider (or rather the deere image placeholder) just below it? this way, we have a splash and quickly give the user a peek into the mixxx UI.

I originally planned to integrate the slider into the splash itself, but if the other Mixxx devs prefer to put the slider below the menu bar we could do that as well.

By the way, this is how the front page looks now:

![2020-05-31-18-58-16_1366x768](https://user-images.githubusercontent.com/1834516/83358095-620ef200-a371-11ea-9f1d-66165adf2d49.png)

I've got to check how spread out the navbar items more evenly, maybe with a flexbox.

daschuer commented:
Thank you for working on this. I like the Idea to change the website from time to time to keep it interesting. It must not be perfect in this sense, so go ahead.   

Her just my two cents:

The screenshot on top of the background looks misplaced. The blurry picture behind the Mixxx logo alone works for me. The blurry picture can even be optimized by showing a laptop running Mixxx along with third party equipment. 

I would prefer to have the screenshot without a distracting surrounding. Maybe below the menu bar. 

The logo in the menu bar in the title page also looks overdone. Since we are already on her home page, it can be removed.  

"Powerful Features For All DJs" is not that catchy for the main phrase on the home page. 
Can we just remove that line? There are enough headlines on top of the menu bar. 

We need IMHO also emphases the unique selling point of Mixxx in the first paragraph: If something does not fit change it. It is the freedom Mixxx users have, no other DJ tool allows that. This should be unmistakably and hit all DJs that are annoyed by artificial restrictions and ignored feature requests in other tools. 

The download button in the content is redundant, because the button in the header bar is always in sight. 

"Everything You Need, However You DJ"  I think this is even more catchy if we remove "However". 
You appears twice. Maybe we can do better, something like: 
"Everything arranged, You DJ"  .. "Everything at hand, you DJ"

Swiftb0y commented:
> The screenshot on top of the background looks misplaced. The blurry picture behind the Mixxx logo alone works for me. The blurry picture can even be optimized by showing a laptop running Mixxx along with third party equipment.

:+1: 

>  I would prefer to have the screenshot without a distracting surrounding. Maybe below the menu bar.

I pretty much meant the same, just that this would be the spot where the image gallery slider would be integrated.

Swiftb0y commented:
> The screenshot on top of the background looks misplaced. The blurry picture behind the Mixxx logo alone works for me. The blurry picture can even be optimized by showing a laptop running Mixxx along with third party equipment.

:+1: 

>  I would prefer to have the screenshot without a distracting surrounding. Maybe below the menu bar.

I pretty much meant the same, just that this would be the spot where the image gallery slider would be integrated.

radusuciu commented:
This looks really good!

Holzhaus committed:
templates/navbar: Use a flexbox for aligning menu items

Holzhaus committed:
templates/navbar: Improve style of download button in hamburger menu

Holzhaus committed:
pages/index: Remove redundant download button

Holzhaus committed:
pages/index: Move screenshot out of splash container

Holzhaus committed:
static/css/custom: Fix some spacing issues

Holzhaus commented:
Ok, I moved the screenshot out of the splash, but IMHO looked much better before:

![2020-05-31-22-07-14_1366x768](https://user-images.githubusercontent.com/1834516/83361686-30a32000-a38b-11ea-984d-ddf830d0c83f.png)

Holzhaus committed:
Revert "pages/index: Move screenshot out of splash container"

This reverts commit a01169a950dc32f2f8cb0ba98bb30d6646304ce0.

Holzhaus commented:
> The blurry picture can even be optimized by showing a laptop running Mixxx along with third party equipment.

If you have a good-looking high-quality photo of Mixxx being used live I'm all for it!

> The screenshot on top of the background looks misplaced. The blurry picture behind the Mixxx logo alone works for me. [...] I would prefer to have the screenshot without a distracting surrounding. Maybe below the menu bar.

I tried that but unfortunately it looks very ugly IMHO. I suggest we add a separate "Screenshots" page which could also add image captions. Then we don't need to put full-size screenshots on the homepage.

> The logo in the menu bar in the title page also looks overdone. Since we are already on her home page, it can be removed.

We can check if the symbolic white logo looks better, but I'd like to keep the navbar consistent on all pages and the logo also acts as a link to the home page.

> The download button in the content is redundant, because the button in the header bar is always in sight.

Ok, I removed it.

> "Powerful Features For All DJs" is not that catchy for the main phrase on the home page.
> Can we just remove that line? There are enough headlines on top of the menu bar.
> 
> We need IMHO also emphases the unique selling point of Mixxx in the first paragraph: If something does not fit change it. It is the freedom Mixxx users have, no other DJ tool allows that. This should be unmistakably and hit all DJs that are annoyed by artificial restrictions and ignored feature requests in other tools.
> 
> "Everything You Need, However You DJ" I think this is even more catchy if we remove "However".
> You appears twice. Maybe we can do better, something like:
> "Everything arranged, You DJ" .. "Everything at hand, you DJ"

I tried to keep the content changes to a minimum because they can't be easily reviewed due to the large diff. I suggest we tackle that in a separate follow-up PR.

Holzhaus committed:
pages/features: Add cover backgrounds to feature page

Holzhaus committed:
pages/features: Clean up button styles

Holzhaus committed:
pages/features: Move "Open Source" section to front page

Holzhaus committed:
Clean up HTML code

Holzhaus committed:
pages/download: Remove potentially tracking flattr.com JS script

Holzhaus committed:
pages/news: Fix invalid time tag

Holzhaus committed:
Add support for embedding YouTube videos

YouTube videos can be embedded like this:

    {% include "video.html" with video_id="aBcDeFgHiJk" %}

Holzhaus committed:
Clean up HTML code and fix some validation errors

Holzhaus committed:
Improve font loading with "font-display: swap" and preloading

Holzhaus committed:
templates/footer: Move alt attribute to img tag

Holzhaus committed:
Only embed mixxx.js to script to download page (it's used nowhere else)

Holzhaus committed:
Add download button with automatic OS detection to download page

Holzhaus committed:
pages/download: Further improve first section on page

Holzhaus committed:
static/js/video: Improve JS code

Holzhaus committed:
pages/error: Improve JS code

Holzhaus committed:
templates/base: Remove redundant extra_script block

Holzhaus committed:
Move screenshots from "Press" page to dedicated "Screenshots" page

Holzhaus committed:
templates/footer: Reorder Footer item and add GitHub link

Holzhaus committed:
pages/index: Clean up main splash on front page

Holzhaus commented:
I removed the screenshot from the splash site and added a dedicated screenshot page.

Here's how it looks on mobile btw:
![Screenshot_20200602_184429_com duckduckgo mobile android](https://user-images.githubusercontent.com/1834516/83546824-96b1b380-a501-11ea-9078-2cd7af987688.jpg)
![Screenshot_20200602_184435_com duckduckgo mobile android](https://user-images.githubusercontent.com/1834516/83546826-97e2e080-a501-11ea-88e8-9b3b13e6cd0c.jpg)

Holzhaus commented:
Here's how it looks on my laptop:

![2020-06-02-18-51-33_1366x768](https://user-images.githubusercontent.com/1834516/83547305-4ab33e80-a502-11ea-9c12-f1cc6f974fc1.png)

Holzhaus committed:
Shrink navbar image to save bandwidth

Holzhaus committed:
Reduce JPEG quality of cover images to speed up page load

ronso0 commented on static/css/barebones.css:
```diff
@@ -0,0 +1,473 @@
+	max-width: var(--grid-max-width);
+	margin: 0 auto;
+	padding: 20px;
+	text-align: center;
```
```suggestion
	text-align: left;
```
IMO this looks more solid, less frayed out.
![image](https://user-images.githubusercontent.com/5934199/83552189-a6cd9100-a509-11ea-8435-e0f2c5b891ba.png)

Holzhaus replied:
Thanks, but I don't want to modify the `barebones.css` file. Instead I added the `u-align-left` CSS class where appropriate.

ronso0 commented on static/css/barebones.css:
```diff
@@ -0,0 +1,473 @@
+ul,
+ol,
+form {
+  margin-bottom: 2.5rem; }
```
```
p { margin-bottom: 3rem; }
```

Holzhaus replied:
To avoid unintended side-effects I'll use this instead:

```
.grid-container p + h5,
.grid-container p + h6 {
    margin-top: 3rem;
}
```

ronso0 commented on static/css/barebones.css:
```diff
@@ -0,0 +1,473 @@
+h3 { font-size: 3.0rem; line-height: 1.3;  letter-spacing: -.1rem; }
+h4 { font-size: 2.4rem; line-height: 1.35; letter-spacing: -.08rem; }
+h5 { font-size: 1.8rem; line-height: 1.5;  letter-spacing: -.05rem; }
+h6 { font-size: 1.5rem; line-height: 1.6;  letter-spacing: 0; }
```
```suggestion
h6 { font-size: 1.5rem; line-height: 1.6;  letter-spacing: 0; margin-bottom: 1rem; }
```
to bring header & paragraph closer together, in conjunction with
```
p { margin-bottom: 3rem; }
```

ronso0 commented on static/css/custom.css:
```diff
@@ -0,0 +1,465 @@
+}
+
+img.responsive {
+    max-width: 100%;
```
```suggestion
    max-width: 100%;
    margin-bottom: 3rem;
```

ronso0 commented on static/css/custom.css:
```diff
@@ -0,0 +1,465 @@
+
+/* Images */
+img.feature-icon {
+    height:4rem;
```
```suggestion
    height:4rem;
    float: left;
    margin-right: 3rem;
```
looks like this, with `grid-comtainer { align: left; }` above,
though it requires `p, h6 { clear: both; }` and maybe more..
![image](https://user-images.githubusercontent.com/5934199/83554690-7687f180-a50d-11ea-9add-c1e8726789e4.png)

ronso0 commented:
Oh yeah, the page feels much more vivid now! Thanks!

I like the background images in Features, though maybe the section on top should have a blurred background since the the text-shadow should be stronger to supply the same readability as with the regular plain sections.

I left a review with a few remarks on paragraphs & header styling and alignment.

The screenshots page:
We should put full size screenshots there instead of skin previews, also show some extreme configurations like 64 samplers, big library + mini decks.
Should we hold this page back and do add it completely in a separate PR?

Swiftb0y commented:
The feature page looks really good now :+1: , though there are a couple of things I'd still change:
1. The text shadow looks pretty good when I test it with chrome, but on my firefox (68.8) the gradient cuts of very abruptly
![Screenshot from 2020-06-02 21-16-26](https://user-images.githubusercontent.com/12380386/83560573-bbfcec80-a516-11ea-87af-b2f897620c76.png)
2. I'd move the skin section above the sampler section. otherwise the user might get a bad impression of the UI when they see screenshots from different skins before knowing there are different skins.

On the screenshot page, I'd leave the preview shots where they are so the user quickly sees every skin but then the hyperlink of each screenshot should point to a high quality version. And more screenshots of the different layouts would be nice.

Holzhaus committed:
pages: Left-align half-page grid containers

Holzhaus committed:
Rework grid-container feature boxes and icons

This is according to ronso0's comments on #97:

    https://github.com/mixxxdj/website/pull/97#discussion_r434072025

Holzhaus committed:
static/css/custom: Add bottom margin for images in grid-containers

Holzhaus committed:
pages/features: Update some feature screenshots

Holzhaus commented:
This is how it looks now:

![2020-06-02-23-35-12_1366x768](https://user-images.githubusercontent.com/1834516/83572394-caa0cf00-a529-11ea-8ec6-74f5d44b148c.png)

Holzhaus commented:
> 1. The text shadow looks pretty good when I test it with chrome, but on my firefox (68.8) the gradient cuts of very abruptly
>    ![Screenshot from 2020-06-02 21-16-26](https://user-images.githubusercontent.com/12380386/83560573-bbfcec80-a516-11ea-87af-b2f897620c76.png)

Maybe a bug in the Firefox rendiering engine, I don't know how I could fix that.

> 2. I'd move the skin section above the sampler section. otherwise the user might get a bad impression of the UI when they see screenshots from different skins before knowing there are different skins.

IMHO we should use the default skin (LateNight PaleMoon) for *all* screenshots and only show other skins on the Screenshots page or the skin section. As a little preview and because of the hardcoded black background of the sampler screenshot I updated some images on the feature page now.

> On the screenshot page, I'd leave the preview shots where they are so the user quickly sees every skin but then the hyperlink of each screenshot should point to a high quality version. And more screenshots of the different layouts would be nice.

I agree, but that can be taken care of in a separate PR, preferably when LateNight has been finalized.

Holzhaus committed:
pages/contact: Improve layout of contact page

Holzhaus committed:
navbar: Move "Get Involved" to "Support & Community" menu

Holzhaus committed:
navbar: Rename "Contact" to "Contact & Team"

Holzhaus committed:
static/js/download.js: Disable main direct download button for Ubuntu

Holzhaus committed:
static/js/download.js: Copy onlick handler on central download button

Holzhaus committed:
pages/contact: Update developer/contributor lists to the 2.2 release

Holzhaus committed:
pages/features: Add shadows to feature screenshots

Holzhaus committed:
templates/video: Improve video template

Holzhaus committed:
pages/screenshots: Use semantic HTML and add shadows

Holzhaus committed:
pages/news: Add blogpost regarding website redesign

Holzhaus commented:
I [added a small blogpost](https://github.com/mixxxdj/website/pull/97/commits/97ca0a279f08a255fe34dbd1cedc771642cafd2d) and made some minor improvements here and there. If there are no major problems I think we can merge now. It's always possible to add improvements later on.

Holzhaus committed:
Scale down images in new posts if they exceed parent container width

daschuer commented:
Can we replace "About" with "Discover" or something. "About" reminds me to the normally boring about box in most applications. I did not expect real content there.

ronso0 commented:
:+1: for "Discover"

One thing I'd like to have adjusted is the letter-spacing, this is th eonly regression I notice. The new font with default letter-spacing is too condensed for continous text at small size IMO and I find it quite straining to read. Let's increase this a little.
(please check the screenshot 1:1 to judge the difference)

now
![image](https://user-images.githubusercontent.com/5934199/83631970-b34ff900-a59e-11ea-9270-d274e050339e.png)


with `p { letter-spacing: .05rem; }`
![image](https://user-images.githubusercontent.com/5934199/83631915-99161b00-a59e-11ea-92ff-bf1f98d7f67c.png)

ronso0 commented:
Overall this is huge improvement! :tada:

Holzhaus committed:
Rename "About" page to "Discover"

Holzhaus committed:
Increase letter spacing to make reading less straining on the eyes

Holzhaus commented:
Done.

ronso0 commented:
LGTM

If there are no more remarks I'd merge this soonish.

radusuciu commented:
Looks great to me as well, very nicely done!

Small suggestion (non-blocking for the merge IMO):
- these buttons are hard to read. maybe something with more contrast here, eg. solid background?
![image](https://user-images.githubusercontent.com/1108600/83661343-d1a3fd80-a57a-11ea-89e6-32aef6757ce4.png)

ronso0 commented:
@Holzhaus Could you polish the buttons @radusuciu mentioned? I think the site background color with 50% opacity would do, then this is ready to roll.

Holzhaus committed:
static/css/custom: Fix wrong comment in stylesheet

Holzhaus committed:
static/css/custom: Make secondary buttons on cover image semitransparent

Holzhaus commented:
Done.

ronso0 commented:
Great, thank you!

ronso0 commented:
(ha! I'm still not a member of the Website repo...can't merge)
@mixxxdj/developers Ready for merge!

Holzhaus commented:
> (ha! I'm still not a member of the Website repo...can't merge)
> @mixxxdj/developers Ready for merge!

In that case I'll merge myself :sunglasses: 

I don't have permissions to add you as a repo member though.

Be-ing commented:
Thanks for working on this. A few issues I noticed:
The navbar image doesn't show Mixxx... it looks like an old Pioneer CDJ. Does Mixxx even support this device? Where did the image come from? Let's not show a device that Mixxx does not support.

The hamburger menu icon on mobile has a small area that was tricky for me to press, especially because it is at the screen edge and I have a case on my phone.

Be-ing commented:
On the front page, the logo in the navigation bar is redundant with the header logo.

Be-ing commented:
On my phone, the logo in the navbar looks blurry. Why not use the SVG version?

Holzhaus commented:
> The navbar image doesn't show Mixxx... it looks like an old Pioneer CDJ. Does Mixxx even support this device? Where did the image come from? Let's not show a device that Mixxx does not support.

It's a stock image: https://www.pexels.com/de-de/foto/audio-beleuchtung-club-deejay-2111016/
Do we have a good-looking photo with Mixxx that we can use?

> The hamburger menu icon on mobile has a small area that was tricky for me to press, especially because it is at the screen edge and I have a case on my phone.

Can you make a screenshot? I'm not sure what you mean.

> On the front page, the logo in the navigation bar is redundant with the header logo.

True, but I don't think that's a big issue. Maybe it would look better if we switched one of these to the symbolic Mixxx logo without colors?
